### PR TITLE
add rocksdb build support in v2

### PR DESCRIPTION
**Pull request by allthatjazzleo**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 PACKAGES=$(shell go list ./... | grep -v '/simulation')
 PACKAGE_NAME:=github.com/crypto-org-chain/chain-main
-GOLANG_CROSS_VERSION  = v1.16.3
+GOLANG_CROSS_VERSION  = v1.17.3
```
Comment (tomtau) [inline Makefile:3]: the latest Go 1.17 is 1.17.13 which had some security fixes -- can that be used instead?

File: cmd/chain-maind/app/app.go [hunk 2]
```diff
@@ -272,6 +272,11 @@ func newApp(logger log.Logger, db dbm.DB, traceStore io.Writer, appOpts serverty
 	}
 
 	snapshotDir := filepath.Join(cast.ToString(appOpts.Get(flags.FlagHome)), "data", "snapshots")
+	err = os.MkdirAll(snapshotDir, os.ModePerm)
```
Comment (tomtau) [inline cmd/chain-maind/app/app.go:275]: not created by SDK?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+*Jan 22, 2022*
+## v2.1.3
+This version is based on Cosmos SDK 0.42.9.
+### Improvements
+* support rocksdb in makefile
```
Comment (tomtau) [inline CHANGELOG.md:7]: the date seems strange? maybe put as unreleased?

**Review by tomtau — COMMENTED** [review]
ack, perhaps the cross can be bumped to the latest minor Go release with all fixes?

File: cmd/chain-maind/app/app.go (hunk 2 above)
Comment (allthatjazzleo) [reply cmd/chain-maind/app/app.go:275]: for the cosmos sdk in chain-main v2, it is required for rocksdb

**Review by tomtau — CHANGES_REQUESTED** [review]
some error for nix flow: https://github.com/crypto-org-chain/chain-main/actions/runs/3369760998/jobs/5589813079#step:5:6

**Comment by yihuang** [comment]
I think rocksdb is supported in nix building already.

**Pull request closed without merging**